model/mattermost_size_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package model
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost-operator/apis/mattermost/v1alpha1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseProvisionerSize(t *testing.T) {
+
+	provisionerXL6Replicas := SizeProvisionerXLResources
+	provisionerXL6Replicas.App.Replicas = 6
+
+	provisionerXL1Replica := SizeProvisionerXLResources
+	provisionerXL1Replica.App.Replicas = 1
+
+	for _, testCase := range []struct {
+		description  string
+		size         string
+		expectedSize v1alpha1.ClusterInstallationSize
+		error        string
+	}{
+		{
+			description:  "correct size, no replicas",
+			size:         "provisionerXL",
+			expectedSize: SizeProvisionerXLResources,
+		},
+		{
+			description:  "correct size, custom replicas (6)",
+			size:         "provisionerXL-6",
+			expectedSize: provisionerXL6Replicas,
+		},
+		{
+			description:  "correct size, ignore other vals",
+			size:         "provisionerXL-1-6",
+			expectedSize: provisionerXL1Replica,
```

Comment:
Hmm, I believe we may want to explicitly block this so we don't end up with size values stored that have extra ignored characters at the end. Something like checking `len(parts) > 2` so we avoid possibly seeing values like `provisionerXL-3-34-333-3`.

Reply:
Yeah, that was intentional to not make validation to strict in case we want to extend this logic further, but yeah we can adjust validation then.